infra/production/materialize-runtime-env.sh

```
@@ -0,0 +1,198 @@
+    if (( octets[0] == 10 )); then return 0; fi
+    if (( octets[0] == 172 && octets[1] >= 16 && octets[1] <= 31 )); then return 0; fi
+    if (( octets[0] == 192 && octets[1] == 168 )); then return 0; fi
+    die "datasource host is not a private IPv4 address"
+  fi
+
+  [[ ${#host} -le 253 && "$host" =~ ^([a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?$ ]] \
+    || die "datasource host is not an approved DNS name"
+}
+
+escape_single_quotes() {
+  printf '%s' "${1//\\'/\\\\'}"
```

Shellcheck (0.11.0)

[info] Want to escape a single quote? echo 'This is how it'\''s done'.

(SC1003)